update charter (#215)
w3c/editing: CHARTER.md

> **End Date:** 31 May 2021 

Reviewer: The end date is bound to the WebAppsWG charter. 
```suggestion
```

**Confidentiality:** Proceedings are [public](https://www.w3.org/2005/10/Process-20051014/comm.html#confidentiality-levels) 

> **Chairs:** Johannes Wilm (Invited Expert), Grisha Lyukshin (Microsoft)

Reviewer: Chairs doesn't seem right... this isn't a working group. 
```suggestion
**Coordinators:** Johannes Wilm (Invited Expert), Grisha Lyukshin (Microsoft)
```

Author: @marcoscaceres Is that following W3C procedures? We had been informed that taskforces can have their own chairs. Were we incorrectly informed about that?

Reviewer: As per the process document - task forces are just informal things: 

> A Chair may form task forces (composed of group participants) to carry out assignments for the group. The scope of these assignments must not exceed the scope of the group's charter. A group should document the process it uses to create task forces (e.g., each task force might have an informal "charter"). Task forces do not publish technical reports; the Working Group may choose to publish their results as part of a technical report.

We are already kinda violating the process document by allowing the task force to publish technical reports... though we will publish anything the task force produces as part of the Working Group. 

The thing is not to cause confusion - the task force is not a Working Group.

Reviewer 2: The formal W3C Process has very little to say about Task Forces. See the last paragraph of https://www.w3.org/2019/Process-20190301/#ReqsAllGroups.

Based on that "Chair of a task force" is not a defined official term. Neither is "Coordinator of a task force". I'd say using one word or the other is a subjective matter of taste. Using the word chair is familiar, and people will know what it means. But it may also lead to some confusions, as people might assume that the role has some formal powers that it doesn't actually have (other than by delegation from the actual WG's chairs).

Reviewer: > But it may also lead to some confusions, as people might assume that the role has some formal powers that it doesn't actually have (other than by delegation from the actual WG's chairs).

Yeah, that's what I'm worried about too. Thanks @frivoal.

Reviewer 2: I think other WGs have Task Forces which call their coordinators "chairs", (I believe that's the case for Houdini for instance), so there's precedent. Maybe a middle ground to limit confusion while avoiding an unfamiliar term would be to be explicit in the TF's charter that the TF chairs are acting as delegates from the WG chairs, and that in case of doubt or conflict, the WG chairs have the final word (and final responsibility).

Reviewer: Sounds good. I can live with that.

Reviewer 3: I am fine with both approaches but I like the alternative @frivoal is proposing.
Should we then create a "definitions" section where we define what chairs mean in this charter?

Reviewer: I'd just include it inline.

Reviewer 3: done.

**Meeting Schedules:** Teleconferences: On an as-needed basis. Preferably, a minimum of one status meeting per month. Face-to-face meetings: On an as-needed basis. Video Conferences: On an as-needed basis. 

### Abstract

Enabling rich editing experience on the web is currently a challenging task. Reasons are many. For example, lack of requirements in the behavior of the `contenteditable` attribute in HTML, currently, the only browser primitive providing rich editing surface to web developers. Also, lack of support for low level editing APIs that would allow web developers to build rich editing experiences without getting browsers interference in this process.  

Editing Task Force sets out to explore limitations in existing browser primitives, provide use cases for new APIs and suggest solutions either by standardizing of existing behaviors or introducing new APIs. The goal is to facilitate the creation of fully-featured editing systems as well as small editors using JavaScript. 

> The Task Force (TF) is part of the [W3C WebApps Working Group](https://w3c.github.io/webappswg/). The TF works primarily within its editing community on [Github](https://github.com/w3c/editing) and will report the results of its activities back to  [W3C WebApps Working Group](https://w3c.github.io/webappswg/).

Reviewer: ```suggestion
The Task Force (TF) is part of the [W3C WebApps Working Group](https://w3c.github.io/webappswg/). The TF works primarily within its editing community on [Github](https://github.com/w3c/editing) and will report the results of its activities back to  [W3C WebApps Working Group](https://www.w3.org/2019/webapps/).
```

The TF is expected to work on the following efforts until the end of the charter but it is not meant to be an exhaustive list:

- [Async Clipboard API](https://bugs.chromium.org/p/chromium/issues/detail?id=931839) 
- [ContentEditableDisabled](http://w3c.github.io/editing/contentEditableDisabled.html) - ability to disable system UI 
- [EditContext API](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/EditContext/explainer.md) 
- [Highlight API](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/highlight/explainer.md) 
- Native Selection and Caret behaviors 
- [SpellChecker API](https://github.com/w3c/editing/issues/166) 
- [SIP Policy](https://github.com/whatwg/html/issues/4876) 
- [Input Events](https://www.w3.org/TR/input-events-1/) 
- [ContentEditable](https://w3c.github.io/contentEditable/) 

> 

Reviewer: @ylafon + @siusin, some of these are not in our charter, and given that "Task forces do not publish technical reports", how can we include these? Do we need to recharter the Web Apps WG?

Reviewer 2: If anything there is on TR and not in WebApps's charter, I guess that a rechartering (or at least a discussion about it) is appropriate. For those that are just EDs or explainers with no presence on TR, I'd consider these under incubation, which is an informal affair that can happen anywhere. They could be added to the charter now, but I don't think it's particularly urgent to do so until we try to publishing them as FPWD (if we're there already, then yes, they should be added).

Reviewer: > For those that are just ED's with no presence on TR, I'd consider these under incubation, which is an informal affair that can happen anywhere. 

Generally speaking, the preference is that incubations happen at the WICG - but it would be ok to incubate things in the WebApps WG as long as they appropriately marked as incubations (i.e., they be clearly labelled "unofficial" until such time that they are not). I sympathetic that everyone who needs to be part of the discussion is following the w3c/editing repo and associated mailing list, which is why I'm also supportive of incubating within WebApps. 

> They could be added to the charter now, but I don't think it's particularly urgent to do so until we try to publishing them as FPWD (if we're there already, then yes, they should be added).

Yes, that makes sense.

Author: AFAIK, respec takes care of marking a spec as not being official as long as one marks it as an editor's draft and not a working draft beyond FPWD. That's really the distinction there is right? Either it's pre-FPWD in which case it's "unofficial" or it is post-FPWD in which case it starts being recognized by the W3C and then psot-rec when it really is soemthing with the W3C's stamp of approval on. 

Or do we need to make a distinction between official and unoffical pre-FPWD editor's draft?

Reviewer: All we need to do is set `"specStatus": "unofficial"` in the ReSpec config. Same thing we did with `execCommand`.

> Deliverables are expected to eventually be shipped as standards recommendation by the Web Applications Working Group, CSS Working Group or any other group where a draft specification may fall into.

Reviewer: ```suggestion
Deliverables are expected to eventually be shipped as Recommendations by the Web Applications Working Group, CSS Working Group or any other group where a draft specification may fall into.
```

> Deliverables are expected to eventually be shipped as standards recommendation by the Web Applications Working Group, CSS Working Group or any other group where a draft specification may fall into.

Reviewer: @ylafon or @siusin this seems odd with regards to deliverables... is it normal to a group to operate across working groups?

Author: @marcoscaceres In practice this is what we do. An example is the Highlight API which we sent on to the CSSWG rather than the Web Apps WG.

Reviewer: I guess eventually this will be a multi-group deliverable? Won't the CSS WG also ask an an incubation process?

Reviewer 2: The CSS WG has decided to take it on, as an ED at first, to be promoted to a FPWD when sufficiently mature (in other words, it has decided to incubate it). Input form the Editing TF is more than welcome, but I don't think we need to consider it a multi-WG deliverable.

Saying that the deliverables are expected to go on the REC track within the Web Apps WG seems right. We should probably also add a liaison section, mention other WGs (such as CSS, or WHATWG…) with which we expect to discuss topics of common interest, and possibly (though not necessarily) list specific such topics, such as the Highlight API.

Reviewer: > The CSS WG has decided to take it on, as an ED at first, to be promoted to a FPWD when sufficiently mature (in other words, it has decided to incubate it). Input form the Editing TF is more than welcome, but I don't think we need to consider it a multi-WG deliverable.

oh, that's great news. 

> Saying that the deliverables are expected to go on the REC track within the Web Apps WG seems right. We should probably also add a liaison section, mention other WGs (such as CSS, or WHATWG…) with which we expect to discuss topics of common interest, and possibly (though not necessarily) list specific such topics, such as the Highlight API.

Agree. That would also align really nicely with the [discussion we had yesterday with the HTML Editors](https://github.com/w3c/contentEditable/issues/2) about `contentEditable`.

### 3. Participation

> This task force is designed to make participation relatively easy for people not currently involved in the standards process (typically, developers of editing tool libraries) who may not be amenable to signing up to a large in scope discussions that can be found in, for instance, the main Web Applications Working Group.

Reviewer: @ylafon or @siusin This is also a bit problematic from an IPR perspective. We encourage folks to participate, but if we also want them to become members - or, at least, Invited Experts. As written, this seems at odds with the function of the W3C Working Groups (vs community groups, where we are much more liberal).

Author: @marcoscaceres This paragraph was part of the original taskforce charter back from when it was first set up.

Reviewer: Understood, but a lot has changed since then.

Reviewer 2: I believe the people who were previously not part of WebApps and who did join the task force did so as Invited Experts of the WG (not 100% sure this has been renewed through the various rechartering, but that was at least the intention initially). The more focused scope is relevant in terms of attention span, but certainly the IPR regime that applies has to be that of the WG as a whole.

Author: That is true. I and several others were made the Invited Expert of some WG, even though we had no clue about what that WG was about or working on and we would not show up to any of its meetings.

Reviewer: @johanneswilm, as external folks engage with the work, encourage them to become Invited Experts (the IPR bot will yell at them regardless if they try to make any pull requests, but if you see someone making substantive suggestions in bug, it's important to get them to join the WG as Invited Experts). The Chairs would be really supportive of that.

Reviewer 3: @marcoscaceres we wanted to remove any barriers in people's participation. I agree that it is fair for people to become Invited Experts if they are ready to commit significant time and effort into editing discussions. (by the way, how do people become Invited Experts?). That said, do people need to go through the process if they just wanted to dial-in or leave comments/open issues in github?

Reviewer: > by the way, how do people become Invited Experts?

It's pretty easy: the Editor's nominate them to the Chairs/Team (team-webapps@w3.org). Then they just fill out a form. 
https://www.w3.org/participate/invited-experts/

It's a quick process. 

> That said, do people need to go through the process if they just wanted to dial-in or leave comments/open issues in github?

Generally no... but it depends on what kind of contribution they are making. If they are adding something that is significant/substantive ("let's add a feature X! it would work like this: ..." or "browser behavior should follow this model: ...") then yes, they MUST become an Invited Expert - what you need to be on the lookout for is for someone claiming "this was my idea!" (because they could have a patent on it). 

If they are just fixing typos, examples, etc. then the do not need to be an Invited Expert. If unsure, ask the Chairs - and be overly cautions.

Reviewer 3: @marcoscaceres understood, thank you. We'll update the wording in a bit.

Reviewer 3: wording has been updated. Let us know if it works for you.

The group encourages questions, comments and all technical discussions on its public [mailing list](https://lists.w3.org/Archives/Public/public-editing-tf/) and [document repositories](https://github.com/w3c/editing).

Participants are reminded of the  [W3C's Code of Ethical Conduct](https://www.w3.org/Consortium/cepc/).

 
### 4. Decision Policy

> Editing Task Force group will seek to make decisions when there is consensus. In cases where there is a need to formally produce a group resolution about a particular issue, its chair(s) will put a question about the issue to the group and gather responses (including any formal objections to formally publishing a document into the appropriate WG); then, after due consideration of all the responses, the Chair(s) will record a group resolution (possibly after a formal vote and also along with responding to any formal objections). 

Reviewer: ```suggestion
The Editing Task force operates under the WebApps Working Group's [decision policy](https://www.w3.org/2019/05/webapps-charter.html#decisions) and adheres to the WebApps WG [working mode](https://www.w3.org/2019/05/webapps-charter.html#working-mode).
```

Reviewer 2: Sounds good. I would add that the TF's Chairs (or coordinators, depending on what we end up calling them) are generally expected to fulfill the role of the Chair listed in the decision policy and working mode, by delegation from the WG chairs (or if they're not, and we expect the actual WG chairs to be the only ones doing that, be clear about it too).

Reviewer 3: Just to clarify this section. These decisions are not about whether something should become a spec draft but rather a shape of a proposal or a specific approach before we propose it to the appropriate WG.

Reviewer: Ok, make sure that's clear when this is updated.

**End Date:** 31 May 2021 

**Confidentiality:** Proceedings are [public](https://www.w3.org/2005/10/Process-20051014/comm.html#confidentiality-levels) 

**Chairs:** Johannes Wilm (Invited Expert), Grisha Lyukshin (Microsoft)

**Meeting Schedules:** Teleconferences: On an as-needed basis. Preferably, a minimum of one status meeting per month. Face-to-face meetings: On an as-needed basis. Video Conferences: On an as-needed basis. 

### Abstract

>   

Reviewer 3: should we kill these lines? it seems that it inserted an extra line below

 

TF is leveraging the existing editing [GitHub](https://github.com/w3c/editing) repo with well-known locations and history of discussions on a variety of editing-related topics in order to “incubate” new editing-specific standards.

 

This charter is intended to reflect on the current direction of the TF group, so that there is common agreement. It may be altered at any point in order to reflect new priorities or work items. 

  
  

### 1. Scope

The scope of Editing TF covers multiple aspects of editing, which may  include:

- Textual input and text manipulation 
- Text editing related events 
- Selection 
- Clipboard 
- Spellcheck and grammar checking 
- Reusing or discarding parts of or entire pre-existing editing APIs to serve as primitives of editing systems 
- Standardization of under-specified existing editing related features 
- Highlighting parts of text ranges without inducing DOM mutations. 

 

Experts who are contributing members of the task force will contribute to documents produced by the group, as described in the section on deliverables below. 

  
  

### 2. Deliverables

This Task Force will

>  

Reviewer 3: I am not sure if you are seeing the same. But I am seeing extra lines when it is rendered on github page.

Author: like here [1]? Looks fine to me now.

[1] https://github.com/w3c/editing/blob/update-charter/CHARTER.md
